app/src/main/java/org/wikipedia/talk/TalkTopicsViewModel.kt

```diff
@@ -0,0 +1,257 @@
+        // have given us a non-Talk page, and we need to prepend the correct namespace.
+        if (pageTitle.namespace.isEmpty()) {
+            pageTitle.namespace = TalkAliasData.valueFor(pageTitle.wikiSite.languageCode)
+        } else if (pageTitle.isUserPage) {
+            pageTitle.namespace = UserTalkAliasData.valueFor(pageTitle.wikiSite.languageCode)
+        } else if (pageTitle.namespace() != Namespace.TALK && pageTitle.namespace() != Namespace.USER_TALK) {
+            // defer resolution of Talk page title for an API call.
+            resolveTitleRequired = true
+        }
+
+        uiState.value = UiState.UpdateNamespace(pageTitle)
+
+        viewModelScope.launch(handler) {
+            if (resolveTitleRequired) {
+                val siteInfoResponse = withContext(Dispatchers.IO) {
+                    ServiceFactory.get(pageTitle.wikiSite).getPageNamespaceWithSiteInfo(pageTitle.prefixedText)
+                }
+                resolveTitleRequired = false
+                siteInfoResponse.query?.namespaces?.let { namespaces ->
+                    siteInfoResponse.query?.firstPage()?.let { page ->
+                        // In MediaWiki, namespaces that are even-numbered are "regular" pages,
+                        // and namespaces that are odd-numbered are the "Talk" versions of the
+                        // corresponding even-numbered namespace. For example, "User"=2, "User talk"=3.
+                        // So then, if the namespace of our pageTitle is even (i.e. not a Talk page),
+                        // then increment the namespace by 1, and update the pageTitle with it.
+                        val newNs = namespaces.values.find { it.id == page.namespace().code() + 1 }
+                        if (page.namespace().code() % 2 == 0 && newNs != null) {
+                            pageTitle.namespace = newNs.name
+                        }
+                    }
+                }
+            }
+
+            val discussionToolsInfoResponse = async { ServiceFactory.get(pageTitle.wikiSite).getTalkPageTopics(pageTitle.prefixedText) }
+            val lastModifiedResponse = async { ServiceFactory.get(pageTitle.wikiSite).getLastModified(pageTitle.prefixedText) }
+            val watchStatus = withContext(Dispatchers.Default) {
+                    if (!sidePanel) ServiceFactory.get(pageTitle.wikiSite)
+                        .getWatchedStatus(pageTitle.prefixedText).query?.firstPage()!! else MwQueryPage()
+            }
+
+            threadItems.clear()
+            threadItems.addAll(discussionToolsInfoResponse.await().pageInfo?.threads ?: emptyList())
+
+            isWatched = watchStatus.watched
+            hasWatchlistExpiry = watchStatus.hasWatchlistExpiry()
+
+            uiState.value = UiState.LoadTopic(pageTitle, threadItems, lastModifiedResponse.await())
+        }
+    }
+
+    val sortedThreadItems get(): List<ThreadItem> {
```

Comment:
This field uses a getter, which performs some pretty heavy operations every time `get()` is called, and it's called multiple times, for every time we `bind()` each topic. Can we pre-compute the `sortedThreadItems` only when the sortMode or searchQuery changes?

Reply:
You're correct. Done.